pkg/utils/kubernetes/kubernetes.go

```diff
@@ -195,23 +194,26 @@ func WaitUntilLoadBalancerIsReady(ctx context.Context, kubeClient kubernetes.Int
 		}
 		return retry.Ok()
 	}); err != nil {
-		fieldSelector := client.MatchingFields{
-			"involvedObject.kind":      "Service",
-			"involvedObject.name":      name,
-			"involvedObject.namespace": namespace,
-			"type":                     corev1.EventTypeWarning,
+		const eventsLimit = 2
+
+		lbService := &corev1.Service{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "Service",
+			},
```

Comment:
Hmm, isn't there another way so that we don't need to explicitly set the `TypeMeta`? I don't have a concrete suggestion but maybe @timebertt has?

Reply:
One could use the client's Scheme to get the `GVK` of the object. E.g.:
```go
gvk, err := apiutil.GVKForObject(obj, c.Scheme())
```
Or directly use the scheme
```go
gvk, isUnversioned, err := c.Scheme().ObjectKinds(obj)
```

Reply:
I followed this approach but ended up in more error handling etc. I chose an alternative and hope you're fine with it 👀

Reply:
Just a minor concern. To my understanding now `FetchEventMessages` relies that `GetLoadBalancerIngress` func will refetch the svc and in this way populate its kind? It is also odd that `GetLoadBalancerIngress` accepts a `corev1.Service` and it does a refetch - with this signature a caller might fetch in advance the svc, call `GetLoadBalancerIngress` and expect that there will be no redundant get call. With the old signature that accepts only the namespace and the name, it is clear the a get request will be issued.

Reply:
Good point @ialidzhikov. I had a similar discussion with @rfranzke about this [here](https://github.com/gardener/gardener/pull/3353#discussion_r553443176).

Reply:
As an alternative I can revert the changes on `GetLoadBalancerIngress` and let it return the `Service` object i.e.:

```
GetLoadBalancerIngress(ctx context.Context, c client.Client, namespace, name string) (string, service *corev1.Service, error)
```

Reply:
I updated the doc string of the function accordingly. Please let me know in case we need a different approach.